include/eve/module/real/core/function/regular/simd/x86/abs.hpp

```diff
@@ -17,38 +17,85 @@
-  // 512 bits implementation
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE wide<T, N, x86_512_> abs_(EVE_SUPPORTS(avx512_), wide<T, N, x86_512_> const &v) noexcept
+  // -----------------------------------------------------------------------------------------------
+  // Masked case
+  template<conditional_expr C, real_scalar_value T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE
+  wide<T, N, ABI> abs_(EVE_SUPPORTS(sse2_), C const &cx, wide<T, N, ABI> const &v) noexcept
   {
-    constexpr auto c = categorize<wide<T, N, x86_512_>>();
+    if constexpr( !ABI::is_wide_logical )
+    {
+      auto m_abs = [&](auto m, auto src, auto a0)
+      {
+        constexpr auto c = categorize<wide<T, N, ABI>>();
 
-         if constexpr ( c && category::unsigned_  ) return v;
-         if constexpr ( c == category::float64x8  ) return _mm512_abs_pd(v);
-    else if constexpr ( c == category::float32x16 ) return _mm512_abs_ps(v);
-    else if constexpr ( c == category::int64x8    ) return _mm512_abs_epi64(v);
-    else if constexpr ( c == category::int32x16   ) return _mm512_abs_epi32(v);
-    else if constexpr ( c == category::int16x32   ) return _mm512_abs_epi16(v);
-    else if constexpr ( c == category::int8x64    ) return _mm512_abs_epi8(v);
+              if constexpr(c && category::unsigned_ ) return if_else(cx,a0,src);
+        else  if constexpr(c == category::float32x16) return _mm512_mask_abs_ps   (src,m,a0);
+        else  if constexpr(c == category::float64x8 ) return _mm512_mask_abs_pd   (src,m,a0);
+        else  if constexpr(c && category::float_    ) return if_else(cx,a0,src);
+        else  if constexpr(c == category::int64x8   ) return _mm512_mask_abs_epi64(src,m,a0);
+        else  if constexpr(c == category::int64x8   ) return _mm256_mask_abs_epi64(src,m,a0);
+        else  if constexpr(c == category::int64x8   ) return _mm_mask_abs_epi64   (src,m,a0);
+        else  if constexpr(c == category::int32x16  ) return _mm512_mask_abs_epi32(src,m,a0);
+        else  if constexpr(c == category::int32x8   ) return _mm256_mask_abs_epi32(src,m,a0);
+        else  if constexpr(c == category::int32x4   ) return _mm_mask_abs_epi32   (src,m,a0);
+        else  if constexpr(c == category::int16x32  ) return _mm512_mask_abs_epi16(src,m,a0);
+        else  if constexpr(c == category::int16x16  ) return _mm256_mask_abs_epi16(src,m,a0);
+        else  if constexpr(c == category::int16x8   ) return _mm_mask_abs_epi16   (src,m,a0);
+        else  if constexpr(c == category::int8x64   ) return _mm512_mask_abs_epi8 (src,m,a0);
+        else  if constexpr(c == category::int8x32   ) return _mm256_mask_abs_epi8 (src,m,a0);
+        else  if constexpr(c == category::int8x16   ) return _mm_mask_abs_epi8    (src,m,a0);
+      };
+
+      return mask_op( x86_512_{}, cx, eve::abs, m_abs, v );
+    }
+    else
+    {
+      return mask_op( cpu_{}, cx, eve::abs, v);
+    }
```

Comment:
I thought mask_op won't be a thing, or did I misunedrstand?

Reply:
It'll, this PR is still incomplete